src/Microsoft.OData.Client/Serialization/HttpClientRequestMessage.cs

```diff
@@ -541,7 +513,7 @@ protected virtual void Dispose(bool disposing)
                     ((IDisposable)response).Dispose();
                 }
 
-                _client.Dispose();
+                _abortRequestCancellationTokenSource.Dispose();

```

Comment:
We don't need to dispose the HttpClient is we didn't use the HttpClientFactory to initialize it?

Reply:
Since we made the default `HttpClient` static, we should not dispose it since it will be re-used by different requests, and its lifetime (if it ever gets created) will be the process's lifetime.

Reply:
From the perspective of isolating requests, would it not be better to new up `HttpClient` instead of using a static instance?
Or at least have PooledConnectionLifetime set - https://learn.microsoft.com/en-us/dotnet/fundamentals/networking/http/httpclient-guidelines#recommended-use

Reply:
@gathogojr that was my initial implementation. But you and @wachugamaina  suggested that I should make the default client static instead. I think having a static client is a better default than a new instance per request. Creating a new client each time, means a new connection each time, this results in higher latencies and more resource utilization. Using the `PooledConnectionLifetime` seems like a good compromise, but how do we decide what the appropriate value is for most users? If we pick a default connection lifetime, then I think it's not any better than picking none and letting the user decide what it should be, but that would be an extra configuration that we'd be exposing to the user. If customers want better control of HttpClient lifetimes, they should provide their `IHttpClientFactory` in my opinion.

As for isolating requests, I'm not sure I understand the scenario. I don't see how sharing the client would cause conflicts between requests. Note that we do allow some request-specific settings like timeouts and cancellation which can be isolated despite sharing the client.